contracts/contracts/strategies/balancer/BaseAuraStrategy.sol

```diff
@@ -0,0 +1,159 @@
+     * well within that abstraction.
+     * @param _rewardTokenAddresses Address of BAL & AURA
+     * @param _assets Addresses of supported assets. MUST be passed in the same
+     *                order as returned by coins on the pool contract, i.e.
+     *                WETH, stETH
+     * @param _pTokens Platform Token corresponding addresses
+     * @param initConfig additional configuration
+     */
+    function initialize(
+        address[] calldata _rewardTokenAddresses, // BAL & AURA
+        address[] calldata _assets,
+        address[] calldata _pTokens,
+        InitConfig calldata initConfig
+    ) external onlyGovernor initializer {
+        auraRewardPoolAddress = initConfig.auraRewardPoolAddress;
+        auraRewardStakerAddress = initConfig.auraRewardStakerAddress;
+        auraDepositorPTokenId = initConfig.auraDepositorPTokenId;
+        pTokenAddress = _pTokens[0];
+        maxWithdrawalSlippage = 1e15;
+        maxDepositSlippage = 1e15;
+        balancerPoolId = initConfig.balancerPoolId;
+        IERC20[] memory poolAssets = getPoolAssets();
+        uint256 assetsLength = _assets.length;
+        require(
+            poolAssets.length == assetsLength,
+            "Pool assets and _assets should be the same length."
+        );
+        for (uint256 i = 0; i < assetsLength; ++i) {
+            (address asset, ) = fromPoolAsset(address(poolAssets[i]), 0);
+            require(
+                _assets[i] == asset,
+                "Pool assets and _assets should all have the same numerical order."
+            );
+            poolAssetsMapped.push(address(poolAssets[i]));
+        }
+
+        super._initialize(
+            initConfig.platformAddress,
+            initConfig.vaultAddress,
+            _rewardTokenAddresses,
+            _assets,
+            _pTokens
+        );
+        _approveBase();
+    }
+
+    function _lpDepositAll() internal virtual override {
+        uint256 bptBalance = IERC20(platformAddress).balanceOf(address(this));
+        IERC4626(auraRewardPoolAddress).deposit(bptBalance, address(this));
+    }
+
+    function _lpWithdraw(uint256 numBPTTokens) internal virtual override {
+        IRewardStaking(auraRewardPoolAddress).withdrawAndUnwrap(
+            numBPTTokens,
+            true // also claim reward tokens
+        );
+    }
+
+    function _lpWithdrawAll() internal virtual override {
+        uint256 bptBalance = IERC4626(auraRewardPoolAddress).balanceOf(
+            address(this)
+        );
+
+        IRewardStaking(auraRewardPoolAddress).withdrawAndUnwrap(
+            bptBalance,
+            true // also claim reward tokens
+        );
+    }
+
+    function collectRewardTokens()
+        external
+        virtual
+        override
+        onlyHarvester
+        nonReentrant
+    {
+        // Collect CRV and CVX
+        IRewardStaking(auraRewardPoolAddress).getReward();
+        _collectRewardTokens();
+    }
+
+    function checkBalance(address _asset)
```

Comment:
🟥 The way `checkBalance` is currently implemented is the classic way to get rekt calculating funds in an AMM.

As a pool gets more out of balance, the sum of assets in the pool goes up. This allows an attacker to provide us with a falsely inflated view of our assets, allowing an attacker to cause a rebase that is too high. We *must* use something like curve's virtual price, a number that can only go up, and is immune to distortion.

🟥 I don't see read-only reentrancy protection. Do we absolutely know that the pool that we are using does not need this? Even if so, there should dire warnings on this file that it is only compatible with this kind of pool. 

🟧  This allows an attacker to manipulate the mix of coins returned by OETH's redeem function. While not critical like the others, and maybe a good design decision (It means that in a non-manipulating case, we have more accurate numbers.) We should probably doc pro and cons of the different approaches here.

🟧 If I'm reading this right, this does a lot of reads / fairly gas inefficient. Checking the value of a strategy is basically the only strategy function that gas matters, because it affects rebasing, minting, and redeeming.